packages/uploadthing/src/internal/runtime.ts

```diff
@@ -0,0 +1,21 @@
+import { FetchHttpClient } from "@effect/platform";
+import * as Layer from "effect/Layer";
+import * as ManagedRuntime from "effect/ManagedRuntime";
+
+import type { FetchEsque } from "@uploadthing/shared";
+
+import type { UTApiOptions } from "../types";
+import { configProvider } from "./config";
+import { withLogFormat, withMinimalLogLevel } from "./logger";
+
+export const makeRuntime = (fetch: FetchEsque, opts?: UTApiOptions) => {
```

Comment:
This type is misleading as we're also passing in `RouteHandlerConfig` in handler.ts.

Either an intersection here or split out a common type?

Reply:
so a common type describing expected Env variables? alternatives would be passing a full configProvider or expecting a JSON Record

Reply:
Json record is probably the most correct, alt a ConfigProvider

Reply:
i'll do a follow-up. couldn't push to your fork